src/Diagnostics.RuntimeHost/Services/SearchService/SearchService.cs

```diff
@@ -30,54 +34,97 @@ public class SearchService : ISearchService
         private string TriggerTrainingUrl;
         private string TriggerModelRefreshUrl;
         private HttpClient _httpClient;
+        private IConfiguration _config;
+        private IHostingEnvironment _env;
+        private bool isEnabled;
 
-        public SearchService()
+        public SearchService(IHostingEnvironment environment, IConfiguration configuration)
         {
+            _config = configuration;
+            _env = environment;
             QueryDetectorsUrl = UriElements.SearchAPI + "/queryDetectors";
             QueryUtterancesUrl = UriElements.SearchAPI + "/queryUtterances";
             TriggerTrainingUrl = UriElements.TrainingAPI + "/triggerTraining";
             TriggerModelRefreshUrl = UriElements.SearchAPI + "/refreshModel";
-            InitializeHttpClient();
+            isEnabled = false;
+            LoadConfigurations();
+            if (isEnabled)
+            {
+                InitializeHttpClient();
+            }
+        }
+
+        public bool IsEnabled()
+        {
+            return isEnabled;
+        }
+
+        public void LoadConfigurations()
+        {
+            if (_env.IsProduction())
+            {
+                isEnabled = Convert.ToBoolean(Registry.GetValue(RegistryConstants.SearchAPIRegistryPath, RegistryConstants.SearchAPIEnabledKey, string.Empty));
+            }
+            else
+            {
+                isEnabled = Convert.ToBoolean(_config[$"SearchAPI:{RegistryConstants.SearchAPIEnabledKey}"]);
+            }
         }
 
-        public Task<HttpResponseMessage> Get(HttpRequestMessage request)
+        public async Task<HttpResponseMessage> Get(HttpRequestMessage request)
```

Comment:
out of curiosity, why this change?  looks like you dont need to await on any statement in this method